storage/simulation/range.go

```diff
@@ -116,8 +130,60 @@ func (r *Range) splitRange(originalRange *Range) {
 	stores := originalRange.getStores()
 	r.Lock()
 	defer r.Unlock()
-	r.desc.Replicas = desc.Replicas
-	r.stores = stores
+	r.desc.Replicas = append([]proto.Replica(nil), desc.Replicas...)
+	for storeID, store := range stores {
+		r.replicas[storeID] = replica{
+			store: store,
+		}
+	}
+}
+
+// prepareActions walks through each replica and determines if any action is
+// required using the allocator.
+func (r *Range) prepareActions() {
+	r.Lock()
+	defer r.Unlock()
+	for storeID, replica := range r.replicas {
+		replica.action, replica.priority = r.allocator.ComputeAction(r.zone, &r.desc)
+		if replica.action == storage.AANoop {
+			replica.rebalance = r.allocator.ShouldRebalance(storeID)
+			replica.priority = 0
+		} else {
+			replica.rebalance = false
+		}
+		r.replicas[storeID] = replica
+	}
+}
+
+// getNextAction returns a single replica with its next action based on the
+// priorities of the each action stored in replicas.
+func (r *Range) getNextAction() (replica, bool) {
```

Comment:
why does it return a replica? seems like it should just return an action, and if there are no replicas, that action is noop